### PR TITLE
Machines buffs and nerfs, recipe unifications, planet biomes

### LINKED ISSUE #1674: Forestry Logs cannot be cut in Mekanism sawmill or Nuclearcraft Manufactory
**Modpack version**
1.80

**Issue**
Tried with Maple, Desert Acacia, Willow, and Teak. The only 3x planks recipe for these logs are the TE sawmill and the IC2 Block Cutting Machine. Thought I'd bring this up to see if it's intended behavior.

### PR DESCRIPTION
**Pull request by Krutoy242**
See commit messages for description of changes.

More description:

- Fixed #1674 
- **AdvRocketry** is t4 mod, so its machines should consume more. It wouldn't affect gameplay much, because they are not usually working in non-stop mode
- **ThermalExpansion cells** needs buff, because it weak in comparsion to RFTools's cells or Mek
- **Lava Heat Exchanger** from EiO was too far in progression (required EiO Alloy Smelter), but produced amusing 80 RF/T. Buffed.
- **RFTools builder** is most terraforming thing in pack. Increasing 100x consumption (for quarry mode) wouldn't affect current uses (like Livingrock farms). But if you want to clear up Nether for 512x512 blocks from bedrock to bedrock, you gonna supply it with ~10 Mil RF/T.
- **Infinity Drill** from Industrial Foregoing have too low requirments for maximum level. I shifted level sizes and energy, so now you need about same GRF to reach 13x13, but you can challenge yourself to reach next levels. Finally place to spend energy from Rainbow Generator.
- **Gas trophy** with any different gases (instead of fixed) will still shows that player used gases in different ways, but not forcing him to build pipelines just for 1 Gas Tank for crafting trophy.
- Remove EnderChest from `ore:chest` to avoid sad situations when you accidentally lose all your obsidian in ME terminal because it autoreplaced Chests to Ender Chests.

Some screenshots:

Morph tool oredict recipe
Morph tool oredict recipe

New recipes for Dynamo
recipes for Dynamo

New consumption rates for EnderIO machines:
consumption rates for EnderIO

Finally more biomes on other planets (Stella):
more biomes

Forestry logs:
Forestry logs

Adv. Rocketry Crystallizer at 100 kRF/t
Rocketry Crystallizer

Demonstration of Gas Trophy crafting function
crafting function

### CODE REVIEW

**Comment by NillerMedDild** [comment]
I look forward to going through the commits, I'm a bit swamped atm though, might be a few days.

**Comment by MorganSimic** [comment]
I am a fan of buffing the TE dynamos because they reallly suck for the most part. But i'm not a fan of recipe. Nerfing the basic dyanmo in such a way is not a good imo because to make them comparable to something like ethylene (grumble) you need resonant dynamos wich is again a lot more gated. I have no idea how the new numbers work out but i'm voting for buff energy, keep recipe. Otherwise it's one step forward and one step back.
Also careful with adjusting the power consumption of the builder. I get the intention but that might break unexpected setups. A little increase is fine, that mach much not really.
Also i would avoid messing with EIO machine consumption. Not sure how the works out with really overpowered Loot Capacitors. Might be fine in the end, if the they match power consumption of NC with 64/64 uprades. It's not fine if they exceed that.
That are just my thoughts on this. Otherwise this is great stuff! So much fixes and behind the scenes QoL, i love it.

**Comment by Krutoy242** [comment]
> Not sure how the works out with really overpowered Loot Capacitors.

Loot capacitors capped for about ~5.0 of power. So, 5.0 capacitor would take about 150 kRF/t



To comparsion 64/64 Rock Crusher consume 520 kRF/t

**Comment by MorganSimic** [comment]
That seems okay, altough the rock crusher is a bad bench mark since it is a special use machine that consumes way more power than the avarage NC machine. Best to compare machine type by machine type, meaning Manufactory to SAG mill and so on.

**Comment by NillerMedDild** [comment]
I would also like to see the Dynamo recipes remain unchanged, at least for the time being. Once people discover they're buffed and start making setups for them, we can look into gating them somewhat, if needed.

File: config/mekanism/BoxBlacklist.txt [hunk 1]
```diff
@@ -11,10 +11,16 @@ astralsorcery:blockstarlightinfuser 0
 astralsorcery:blockwell 0
 astralsorcery:blockattunementrelay 0
 astralsorcery:blockritualpedestal 0
-astralsorcery:blockaltar
-astralsorcery:blockcelestialcrystals
+astralsorcery:blockaltar 0
```
Comment (NillerMedDild) [inline config/mekanism/BoxBlacklist.txt:14]: Not appending a number in the BoxBlacklist.txt means it's a fuzzy match, all meta values. Please undo this change.

File: config/industrialforegoing.cfg [hunk 2]
```diff
@@ -31,19 +31,19 @@ general {
             S:0_poor=0
 
             #  [default: 4000000]
```
Comment (NillerMedDild) [inline config/industrialforegoing.cfg:33]: Why did you change the Infinity Drill?

File: config/rftools/rftools.cfg [hunk 3]
```diff
@@ -56,28 +56,28 @@ booster {
```
Comment (NillerMedDild) [inline config/rftools/rftools.cfg:56]: I do not approve of this change of rf consumption.

File: scripts/Creative.zs [hunk 4]
```diff
@@ -62,26 +64,79 @@ var twilightForestMasterTrophy = <simple_trophies:trophy>.withTag({
   TrophyVariant:"gold"
 });
 
+
```
Comment (NillerMedDild) [inline scripts/Creative.zs:67]: Could you please explain what this is?

File: scripts/Creative.zs [hunk 5]
```diff
@@ -218,7 +273,7 @@ var twilightForestMasterTrophy = <simple_trophies:trophy>.withTag({
 	[hvSolar, ultProvider, ultProvider, anglesite, anglesite, anglesite, ultProvider, ultProvider, hvSolar], 
 	[hvSolar, ultProvider, <ic2:te:75>, <ic2:te:22>, <nuclearcraft:fusion_core>, <ic2:te:22>, <ic2:te:75>, ultProvider, hvSolar], 
 	[hvSolar, anglesite, <ic2:te:22>, ultCell, ultCatalyst, ultCell, <ic2:te:22>, anglesite, hvSolar],
```
Comment (NillerMedDild) [inline scripts/Creative.zs:275]: Is this changing the recipe from requiring the full generator to only half?

File: scripts/Mekanism.zs [hunk 6]
```diff
@@ -269,4 +268,7 @@ for input, output in itemsToUnify {
 	rh(<mekanism:oreblock:2>);
 	rh(<mekanism:walkietalkie>);
 
-print("--- Mekanism.zs initialized ---");
\ No newline at end of file
+# Tier applying (for AE extended recipe)
```
Comment (NillerMedDild) [inline scripts/Mekanism.zs:271]: Please remove this, as it will void the contents of the machine you're upgrading.

File: scripts/OreDict.zs [hunk 7]
```diff
@@ -663,6 +663,10 @@ for item in pressurePlates {
 	
 	Door.addItems(DoorToAdd);
 
+# *======= Remove ender chest =======*
```
Comment (NillerMedDild) [inline scripts/OreDict.zs:666]: Ender Chest is a chest, please remove this :)

File: scripts/ThermalExpansion.zs [hunk 8]
```diff
@@ -186,4 +187,32 @@ print("--- loading ThermalExpansion.zs ---");
...
+recipes.removeByRecipeName("thermalexpansion:cell_14");
+recipes.removeByRecipeName("thermalexpansion:cell_18");
+recipes.removeByRecipeName("thermalexpansion:tank_18");
+recipes.removeByRecipeName("thermalexpansion:tank_16");
+recipes.removeByRecipeName("thermalexpansion:tank_17");
+recipes.removeByRecipeName("thermalexpansion:tank_14");
+recipes.removeByRecipeName("thermalexpansion:tank_15");
+
+// # Dynamo harder recipes
+function remakeDynamo(name as string, item as IItemStack, i1 as IIngredient, i2 as IIngredient){
```
Comment (NillerMedDild) [inline scripts/ThermalExpansion.zs:203]: Please undo the recipe changes here, at least for the time being. I know I've previously said I liked them, but I agree with the point that the recipes should not be made harder while they remain largely unused.

**Comment by NillerMedDild** [comment]
Sorry for the messy review, still don't really understand how to do it properly :P

The vast majority of these changes look great, thanks!

File: config/mekanism/BoxBlacklist.txt (hunk 1 above)
Comment (Krutoy242) [reply config/mekanism/BoxBlacklist.txt:14]: > it's a fuzzy match, all meta values.

No, its not. 
First, lines without number at the end listed in debug.log file with error spelling "Cant parse line" or something like this.
Second, on current E2E version you still can put box on Altairs or Celestia Crystals, that obiviously shows that blacklist isn't working.

File: config/industrialforegoing.cfg (hunk 2 above)
Comment (Krutoy242) [reply config/industrialforegoing.cfg:33]: Because energy system in E2E is reworked, and some things like Mek reactors have huge output. In this environment, when drill become available you already have power production enough to reach maximum level in instant (10 minutes at 200 kRF/t).
Its not "Fix". Just suggestion.

File: scripts/Creative.zs (hunk 5 above)
Comment (Krutoy242) [reply scripts/Creative.zs:275]: I suggest to add Rainbow Stone instead of rainbow generator. Because it require working Rainbow Generator what harder to make than just generator itself. Through, to make one stone you not need to bother with fully automatic working generator.
https://i.imgur.com/aTe6RbW.png

File: scripts/Creative.zs (hunk 4 above)
Comment (Krutoy242) [reply scripts/Creative.zs:67]: This is changing Gas Trophy.

1. When you watching recipe (R key) you see scrolling gases of many kinds
2. Also you see description that gas trophy crafted with 9 different gases
3. When you place 9 gas tanks in grid crafting function checks for all gases would be different
4. If gases different, it shows trophy as output

There is example how its looks:
https://i.imgur.com/NBFQpsK.gif

File: scripts/Mekanism.zs (hunk 6 above)
Comment (Krutoy242) [reply scripts/Mekanism.zs:271]: This is only problem? I can write function that saves all NBT tags and just changes Tier

**Comment by NillerMedDild** [comment]
Alright so many comments are scattered to hell, gonna write them here instead:
* Do not remove the EnderChest from the Chest oredict
* Do not make a Mekanism tier upgrade recipe, it's not the risk
* Using the Rainbow Stone is a good idea, but I do not want to force players to make a working Rainbow Generator setup. Please remove this change.
* I do not want the Gas Trophy recipe change. The Gases I chose are not random.
* Please undo the Infinity Drill changes. The highest tier requires 9223372036854775807 RF, which is insane enough, no need to increase the other ones too.
* Ignore my comment at BoxBlacklist if it doesn't work in it's current state.
